crates/pathfinder/src/sequencer.rs

```diff
@@ -37,17 +47,21 @@ impl Client {
     }
 
     /// Gets block by number.
-    pub async fn block_by_number(&self, block_number: BlockNumberOrTag) -> Result<reply::Block> {
+    pub async fn block_by_number(
+        &self,
+        block_number: BlockNumberOrTag,
+    ) -> Result<reply::Block, SequencerError> {
         let block_hash = match block_number {
             BlockNumberOrTag::Number(n) => {
                 let resp = reqwest::get(
                     self.build_query("get_block_hash_by_id", &[("blockId", &n.to_string())]),
                 )
                 .await
                 .unwrap();
```

Comment:
Sorry to add new ones at this stage but is this unwrap a left-over? Given the subject of this PR I'd expect this to be handled, or is it a class you are intentionally leaving for later. Surely there could be networking errors which should just become SequencerErrors?

Reply:
Other places don't seem to unwrap on `reqwest::get` return value.

Reply:
This is definitely a bug! Many thanks for spotting!